port/cpl_aws.cpp

```diff
@@ -1746,6 +1758,172 @@ bool VSIS3HandleHelper::GetOrRefreshTemporaryCredentialsForSSO(
     return false;
 }
 
+/************************************************************************/
+/*                  GetCredentialsFromProcess()                         */
+/************************************************************************/
+
+static bool GetCredentialsFromProcess(const std::string &osCredentialProcess,
+                                      std::string &osSecretAccessKey,
+                                      std::string &osAccessKeyId,
+                                      std::string &osSessionToken)
+{
+    CPLDebug(AWS_DEBUG_KEY, "Executing credential_process: %s",
+             osCredentialProcess.c_str());
+
+    CPLStringList aosOptions;
+    aosOptions.SetNameValue("TIMEOUT", "30");
+
+    FILE *fp = popen(osCredentialProcess.c_str(), "r");
```

Comment:
Should we use CPL-family functions in cpl_spawn.h such as CPLPipeRead for multi-platform compatibility instead of directly calling POSIX API here?